src/pkg/cluster/injector.go

```diff
@@ -384,13 +407,22 @@ func (c *Cluster) createService(ctx context.Context) (*corev1.Service, error) {
 // buildInjectionPod return a pod for injection with the appropriate containers to perform the injection.
 func (c *Cluster) buildInjectionPod(node, image string, payloadConfigmaps []string, payloadShasum string) (*corev1.Pod, error) {
 	executeMode := int32(0777)
+
+	// Create a SHA-256 hash of the image name to allow unique injector pod names.
+	// This prevents collisions where `zarf init` is ran back to back and a previous injector pod still exists.
+	hasher := sha256.New()
+	if _, err := hasher.Write([]byte(image)); err != nil {
+		return nil, err
+	}
+	hash := hex.EncodeToString(hasher.Sum(nil))[:8]
```

Comment:
If we select the same image as the first Zarf init, wouldn't we end up with the same sha making the collision still possible?

Reply:
The Pod should eventually be deleted so this should not be an issue right? Unless you run them immediatly after each other.

Reply:
Yes although running Zarf init right after each other is what is breaking the upgrade test. The bigger problem seems to be that the 0 second grace period isn't taking effect. Running locally the injector pod is terminating for around 20 seconds now.

I also don't think it's an unreasonable use case for a real user to run into. 
zarf init -> realize they forgot to select git server -> zarf init --components=git-server

Reply:
Grace period of 0 is fine actually, likely we are going to refactor this anyway

Reply:
Setting grace period to 0 does not fix the error in CI

https://github.com/defenseunicorns/zarf/pull/2615#issuecomment-2161701286

I have yet to see the error occur in this PR after running multiple pipelines back to back

Reply:
Added UUID instead of sha for better collision resistance